src/utils/kernel.rs

```diff
@@ -49,6 +49,22 @@ where
         let channel_shape = self.filter.output_shape(channelless_input);
         Tensor::from_shape_vec(channel_shape, output).unwrap()
     }
+
+    pub fn weights(&self) -> Tensor<E> {
+        self.weights.clone()
+    }
+
+    pub fn bias(&self) -> f64 {
+        self.bias.value()
+    }
+
+    pub fn weights_mut(&mut self) -> &mut Tensor<E> {
```

Comment:
I feel like `weights_mut` and `set_bias` both attempt to do the same thing but are implemented in different ways, choosing one approach might be cleaner

Reply:
Agreed, these methods were some of the ones I was least confident in. How would you suggest having a single approach, though, since kernel weights is a value tensor, whereas kernel bias is a single value?

Reply:
By one method i mean either `weights_mut` and `bias_mut` or `set_weights` and `set_bias`

Reply:
Ok I see what you mean. Would it be fine to have something like the following in convolution.rs:

```
set_weights() {
  for each kernel in kernels {
    kernel.set_weights()
  }
}
```

I avoided this because of the repeated name `set_weights`, so I'm leaning towards `weights_mut` and `bias_mut` as you suggested.